### PR TITLE
Fix some issues in McCall Q-learns

### LINKED ISSUE #310: [mccall_q] equation (2) in remark and a sentence to be completed
Hi @thomassargent30 thanks for writing this beautiful and interesting lecture on the McCall model and q learning in reinforcement learning. I love both of them.

As discussed with @jstac  we've found two typos in lecture [mccall_q](https://python.quantecon.org/mccall_q) and wish you can have a look when you are available.

One is about the location of ``equation (2)`` mentioned in the following remark of subsection [q-learning](https://python.quantecon.org/mccall_q.html#q-learning):
``**Remark:** Notice that $\widetilde{TD}$ associated with an optimal Q-table defined in equation (2) automatically above satisfies  $\widetilde{TD}=0$ for all state-action pairs.  Whether a limit of our Q-learning algorithm converges to an optimal Q-table depends on whether the algorithm visits all state-action pairs often enough.``

The other is that there is an incomplete sentence in subsection [employed-worker-cant-quit](https://python.quantecon.org/mccall_q.html#employed-worker-cant-quit):
``But if we increase the numb``.

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #310 

cc @thomassargent30 @jstac

### CODE REVIEW

**Comment by Smit-create** [comment]
@mmcky The CI fails with:
```
ERROR: Execution failed in an unexpected way, clearing staged notebooks: The URI of the staged record no longer exists: /__w/lecture-python.myst/lecture-python.myst/lectures/cross_product_trick.md
ERROR: Couldn't find cache key for notebook file lectures/mccall_q.md. Outputs will not be inserted.
```

Are there any issues in this diff that trigger this error?

**Comment by Smit-create** [comment]
@thomassargent30 I haven't merged this in `main` because of some failures in CI. You can pull this branch to have a look at the changes. I can help you with that.

File: lectures/mccall_q.md [hunk 1]
```diff
@@ -17,32 +17,36 @@ kernelspec:
 
 This lecture illustrates a powerful machine learning technique called Q-learning.
 
-{cite}`Sutton_2018` presents Q-learning and a variety of other statistical learning procedures.
+[[SB18](https://python.quantecon.org/zreferences.html#id29)] presents Q-learning and a variety of other statistical learning procedures.
```
Comment (thomassargent30) [inline lectures/mccall_q.md]: @Smit-create these should remain as {cite} links through bib file

File: lectures/mccall_q.md [hunk 2]
```diff
@@ -17,32 +17,36 @@ kernelspec:
...
+- a recursive version of least squares known as **temporal difference learning**
 
-* a recursive version of least squares known as [temporal difference learning](https://en.wikipedia.org/wiki/Temporal_difference_learning).
 
-This lecture applies a Q-learning algorithm to the situation faced by  a   McCall worker.
+This lecture applies a Q-learning algorithm to the situation faced by  a McCall worker.
+
+This lecture also considers the case where a McCall worker is given an option to quit the current job.
+
+Relative to the dynamic programming formulation of the McCall worker model that we studied in  [quantecon lecture](https://python.quantecon.org/mccall_model.html), a Q-learning algorithm gives the worker less knowledge about
```
Comment (thomassargent30) [inline lectures/mccall_q.md]: @Smit-create   quantecon lecture should remain as a doc link

File: lectures/mccall_q.md [hunk 3]
```diff
@@ -51,69 +55,72 @@ actions that he can take in each state.
 
 An equation  that resembles a  Bellman equation  plays an important role in the algorithm.
 
-It  differs from the Bellman equation for the McCall model that we have seen in {doc}`this quantecon lecture <mccall_model>`
+It  differs from the Bellman equation for the McCall model that we have seen in [this quantecon lecture](https://python.quantecon.org/mccall_model.html)
```
Comment (mmcky) [inline lectures/mccall_q.md]: This needs to be remain as a `doc` link

File: lectures/mccall_q.md [hunk 4]
```diff
@@ -51,69 +55,72 @@ actions that he can take in each state.
...
+- the **Q-function** or **quality function** that is affiliated with any Markov decision problem whose optimal value function satisfies a Bellman equation
+- **temporal difference learning**,  a key component of a Q-learning algorithm
 
-* **temporal difference learning**,  a key component of a Q-learning algorithm
 
-As usual, let's  import some Python modules.
+As usual, let’s  import some Python modules.
 
 ```{code-cell} ipython3
+!pip install quantecon
```
Comment (mmcky) [inline lectures/mccall_q.md:70]: perhaps add `:tag: [hide-output]`

File: lectures/mccall_q.md [hunk 5]
```diff
@@ -51,69 +55,72 @@ actions that he can take in each state.
...
 
-With possible states residing in a finite discrete state space indexed by $\{1,2,...,n\}$, we make an initial guess for the value function of $v\in\mathbb{R}^{n}$ and then iterate on the Bellman equation:
+With possible states residing in a finite discrete state space indexed by $ \{1,2,...,n\} $, we make an initial guess for the value function of $ v\in\mathbb{R}^{n} $ and then iterate on the Bellman equation:
 
 $$
 v^{\prime}(i)=\max \left\{\frac{w(i)}{1-\beta}, c+\beta \sum_{1 \leq j \leq n} v(j) q(j)\right\} \quad \text { for } i=1, \ldots, n
 $$
 
-Let's use  Python   code from {doc}`this quantecon lecture <mccall_model>`. 
+Let’s use  Python   code from [this quantecon lecture](https://python.quantecon.org/mccall_model.html).
```
Comment (mmcky) [inline lectures/mccall_q.md]: `doc` link

File: lectures/mccall_q.md [hunk 6]
```diff
@@ -51,69 +55,72 @@ actions that he can take in each state.
...
 import matplotlib.pyplot as plt
-from tqdm import tqdm
 
 np.random.seed(123)
 ```
 
 ## Review of McCall Model
 
-We begin by reviewing the McCall model described in {doc}`this quantecon lecture <mccall_model>`.
+We begin by reviewing the McCall model described in [this quantecon lecture](https://python.quantecon.org/mccall_model.html).
```
Comment (mmcky) [inline lectures/mccall_q.md]: `doc` link

File: lectures/mccall_q.md [hunk 7]
```diff
@@ -135,11 +142,14 @@ ax.set_ylabel('probabilities')
...
 
-Next we'll compute the worker's optimal value function by iterating to convergence on the Bellman equation.
+Next we’ll compute the worker’s optimal value function by iterating to convergence on the Bellman equation.
 
-Then we'll plot various iterates on the Bellman operator.
+Then we’ll plot various iterates on the Bellman operator.
 
 ```{code-cell} ipython3
+:hide-output: false
+:id: 415c496e
```
Comment (mmcky) [inline lectures/mccall_q.md]: @Smit-create do you know where these `id` are coming from?

File: lectures/mccall_q.md [hunk 8]
```diff
@@ -245,218 +256,190 @@ But value functions  are functions just of states, and not actions.
...
 \end{aligned} 
-$$ (eq:impliedq)
-
-
-Note that the first equation of system {eq}`eq:impliedq` presumes that after  the agent has  accepted an offer, he will not have the objection to reject that same offer in the future.   
-
-These equations are aligned with the Bellman equation for the worker's  optimal value function that we studied in {doc}`this quantecon lecture <mccall_model>`. 
+$$ (eq-impliedq)
 
+Note that the first equation of system [(39.2)](#equation-eq-impliedq) presumes that after  the agent has  accepted an offer, he will not have the objection to reject that same offer in the future.
```
Comment (mmcky) [inline lectures/mccall_q.md]: use `{eq}` role for equation numbering

File: lectures/mccall_q.md [hunk 9]
```diff
@@ -245,218 +256,190 @@ But value functions  are functions just of states, and not actions.
...
 $$
-V(w) = \max_{\textrm{accept},\textrm{reject}} \left\{ Q(w, \text{accept} \right), Q\left(w,\text{reject} \right)\}
+V(w) = \max_{\textrm{accept},\textrm{reject}} \left\{ Q(w, \text{accept} \right), Q\left(w,\text{reject} \right\}
 $$
 
-If we stare at the second equation of system {eq}`eq:impliedq`, we notice that since the wage process is identically and independently distributed over time,
- $Q\left(w,\text{reject}\right)$, the right side of the equation is independent of the current state   $w$.
- 
-So we can denote it as a scalar  
+If we stare at the second equation of system [(39.2)](#equation-eq-impliedq), we notice that since the wage process is identically and independently distributed over time,
```
Comment (mmcky) [inline lectures/mccall_q.md]: use `{eq}` role

File: lectures/mccall_q.md [hunk 10]
```diff
@@ -245,218 +256,190 @@ But value functions  are functions just of states, and not actions.
...
+$$
+Q_r=Q\left(w,\text{reject}\right),\forall w\in\mathcal{W}.
 $$
 
-This fact provides us with an 
+This fact provides us with an
 an alternative, and  as it turns out in this case, a faster way to compute an optimal value function and associated optimal policy for the McCall worker.
 
-Instead of using the  value function iterations that we deployed above, we can instead  iterate to convergence on a version of the second equation in system {eq}`eq:impliedq`  that maps an estimate of  $Q_r$ into an improved estimate $Q_r'$:
+Instead of using the  value function iterations that we deployed above, we can instead  iterate to convergence on a version of the second equation in system [(39.2)](#equation-eq-impliedq)  that maps an estimate of  $ Q_r $ into an improved estimate $ Q_r' $:
```
Comment (mmcky) [inline lectures/mccall_q.md]: use `{eq}` role

File: lectures/mccall_q.md [hunk 11]
```diff
@@ -245,218 +256,190 @@ But value functions  are functions just of states, and not actions.
...
-```{code-cell} ipython3
-print(compute_value_function_two(mcm))
-```
 
 ## From Probabilities  to Samples
 
 We noted  above that  the optimal Q function for our McCall worker satisfies the Bellman equations
 
+
+<a id='equation-eq-probtosample1'></a>
```
Comment (mmcky) [inline lectures/mccall_q.md]: keep this equation as a labelled equation using `$$ (label)` syntax

**Review by mmcky — COMMENTED** [review]
thanks for this PR @Smit-create it would be great if you could

1. keep using the `{eq}` role for referencing equations and maintain the eq label syntax rather than use markdown. 
2. remove any `\tag` from the latex
3. reference other documents using `{doc}` role

**Comment by mmcky** [comment]
@Smit-create I have made some comments and requests. Let me know if you'd like to link up to review.

**Comment by mmcky** [comment]
Have a quick look through this as well

https://manual.quantecon.org/writing/styleguide.html#mathematical-notation

that might help. Please suggest any improvements etc.

**Comment by Smit-create** [comment]
Ahh, these issues are here because I used colab for editing the notebook. I have now tried to fix it locally. Thanks for the review @mmcky

**Comment by Smit-create** [comment]
@thomassargent30 @mmcky The CI is green. Please have a look.

**Pull request merged**